src/workflows/airqo_etl_utils/airqo_utils.py

```diff
@@ -370,13 +373,11 @@ def extract_devices_data(
                         )
                     )
 
-                data["device_number"] = device_number
-                data["device_id"] = device.get("device_id")
-                data["site_id"] = device.get("site_id")
-
-                if device_category == DeviceCategory.BAM:
-                    data["latitude"] = meta_data.get("latitude", None)
-                    data["longitude"] = meta_data.get("longitude", None)
+                data["device_number"] = device.get("device_number", None)
+                data["device_id"] = device.get("device_id", None)
+                data["site_id"] = device.get("site_id", None)
+                data["latitude"] = device.get("latitude", None)
```

Comment:
Thanks @MartinKalema for this PR.
I have a few inquiries:

- Do these latitude and longitude values come from field8 of thingspeak?
- Also generally, are these GPS values coming from ThingSpeak?
- If these GPS values are coming from ThingSpeak, we shall need to handle edge cases accordingly appropriately since it has been known for long that the GPS sensors from our devices are very unstable
- If these GPS values are coming from Device Details (NOT Thingspeak), then we also need to cater for scenarios where some devices do not have these values --- especially those which have not yet been deployed.

**Disclaimer:** The inquiries shared above are based on the high level understanding of things and might be slightly outside the scope of your PR so please feel free to advise as you respond accordingly.

Thanks!

Reply:
The GPS values for all Low Cost devices are coming from the device details.

Why?
 - Only BAM devices seem to have GPS data from ThingSpeak. And also, the previous implementor ensured that GPS data was being fetched from ThingSpeak for the case of BAM devices.  
 - For all API calls to ThinkSpeak i've made related to LOW COST devices, the latitude and longitude values in field8 are "0.0000". This heavily contributed to the decision to just rely on devices data for all LOW COST devices rather than ThingSpeak.
 
 So can i make ThingSpeak a default for BAM devices and Device details a default for LOW_COST devices?

Reply:
@MartinKalema Did you look at jira card OPS-231? While some devices do not have location data, others do. There are screenshots attached to the card.

Reply:
@NicholasTurner23  I've just seen the card and i noticed you are looking at channel data rather than data from the feeds. The feeds have a column called `field8` that contains 8 values that represent;
 - latitude
 - longitude
 - altitude
 - wind_speed
 - satellites
 - hdop
 - device_temperature
 - device_humidity
 - temperature
 - humidity
 - vapor_pressure
This is where the latitude and longitude data are fetched but these values are "0.00000" for all LOW_COST devices. Refer to the image below

![Screenshot_20240624_110320](https://github.com/airqo-platform/AirQo-api/assets/130189437/4da32045-25aa-4e8b-be55-f435872a02b9)

Reply:
Thanks a lot @MartinKalema and @NicholasTurner23 for this amazing thread.

Just one last thing....since only DEPLOYED devices will have corresponding GPS values, how do we deal with the edge case of those devices that have NOT yet been deployed through AirQo Analytics?

**Context:** To answer this question appropriately, you must be knowledgeable about the device DEPLOYMENT process. Always feel free to reach out to @Codebmk  for more details in case you are not yet familiar with the process.

Reply:
> Thanks a lot @MartinKalema and @NicholasTurner23 for this amazing thread.
> 
> Just one last thing....since only DEPLOYED devices will have corresponding GPS values, how do we deal with the edge case of those devices that have NOT yet been deployed through AirQo Analytics?
> 
> **Context:** To answer this question appropriately, you must be knowledgeable about the device DEPLOYMENT process. Always feel free to reach out to @Codebmk for more details in case you are not yet familiar with the process.

Hi @MartinKalema , awaiting your response to my last question here..... just want to hear you out on this edge of devices which INTENTIONALLY do not have Site Details since they are NOT yet deployed. How does the current implementation cater for this? I also left some notes on a JIRA card that might be related to this. Here is the JIRA card: https://airqoteam.atlassian.net/browse/OPS-232

Reply:
@Baalmart, apologies for the delay. Our meeting got rescheduled to today.

I discussed this with Belinda, George, and Gibson. We've decided that for devices that haven't been deployed yet, we'll use static longitude and latitude values for Makerere University since these devices are currently in the lab.

I'll proceed to edit the code accordingly to reflect this decision. Regarding the missing data in ThingSpeak field8, I'll write a report outlining the issue. The hardware team will investigate the root cause further.

Let me know if you have any further questions or if there's anything else we need to address regarding this approach.

Reply:
Thanks @MartinKalema .....

Why not just filter out devices which have not yet been deployed as a first step in our algorithm? So that we just deal with deployed devices ONLY?

In some JIRA card, I shared a query parameter (`active`) of query devices endpoint which enables one to return DEPLOYED devices ONLY. To only return deployed devices, you just have to query (`?active=yes`)